src/nbl/video/COpenGLCommandBuffer.h

```diff
@@ -581,14 +581,7 @@ class COpenGLCommandBuffer final : public IGPUCommandBuffer
         const auto bindingFlags = pipeline->getVertexInputParams().enabledBindingFlags;
         auto vertexBufferBindings = meshBuffer->getVertexBufferBindings();
 
-        for (uint16_t i = 0; i < nbl::asset::SVertexInputParams::MAX_ATTR_BUF_BINDING_COUNT; ++i)
-        {
-            if (bindingFlags & core::createBitmask({ i }))
-            {
-                auto vertexBufferBinding = vertexBufferBindings[i];
-                bindVertexBuffers(i, 1, &vertexBufferBinding.buffer.get(), &vertexBufferBinding.offset);
-            }
-        }
+        bindVertexBuffers(0, nbl::asset::SVertexInputParams::MAX_ATTR_BUF_BINDING_COUNT, &vertexBufferBindings[0].buffer.get(), &vertexBufferBindings[0].offset); // TODO: I'm not sure about pOffset* in this approach
```

Comment:
@Crisspl help plz, does he need to make a temporary copy before firing this off?

Reply:
wait wtf this code even is

Reply:
I think like I would have to create temporary array of offset before and pass it?

Reply:
no, he doesnt need tmp copy, 
but: @AnastaZIuk 
1) you cant just pass pointer to member of struct as a pointer to array of buffers
layout in mem is buf0|offset0|buf1|offset1|buf2|offset2|.... and function  expects it to be buf0|buf1|buf2 obviously.
Same with offsets.
2) im not sure if you can pass null buffers there but should be ok

Reply:
> I think like I would have to create temporary array of offsets before and pass it?

in that sense, yeah of course, because of memory layout
`&vertexBufferBindings[0].buffer.get()` is not a pointer to array of buffers and `&vertexBufferBindings[0].offset` is not a pointer to array of offsets

Reply:
> you cant just pass pointer to member of struct as a pointer to array of buffers
layout in mem is buf0|offset0|buf1|offset1|buf2|offset2|.... and function expects it to be buf0|buf1|buf2 obviously.
Same with offsets.

So he needs to copy to a temporary stack array deinterleave (should probably look at old `COpenGLDriver.cpp` to see how it used to be done)

Reply:
well, maybe then you would like to change the api to take array of binding structs? @devshgraphicsprogramming

Reply:
i havent thought about this before and its very very vulkan-like

Reply:
corrected

Reply:
> 
> 
> well, maybe then you would like to change the api to take array of binding structs? @devshgraphicsprogramming

thou hast my blessings

Reply:
discussed with @Crisspl that there should be an overload for that, but who of us should do this?

Reply:
i think you, it's simple, but IMHO it's optional, Matt's decision whether to do it a all

Reply:
whatever makes sense to you guys, but remember the Vulkan impl will have to build its own little arrays of 16 things anyway.

Reply:
okay I will write an overload